### DISCUSSION TITLE
A little 'rant' about LXQt flaws and questionable decisions

### DISCUSSION THREAD

**Discussion by renegat73** [lxqt/lxqt#2035] [General]
Warning: This thread could be considered as a rant. So if you are one of this typical developers who thinks every criticism is a rant or heresy please don't read further.

I was using LXQt from the very beginning as it appears in the packages of archlinux and I was maintaining quite a couple of systems running it.
I liked the idea of a modern successor of LXDE which I had used till then. But meanwhile I think some of the developers lost their way and implement functions to it which are totally waste while removing totally reasonable patches and functions.

It started when someday after an update of LXQt to 0.14 suddenly every icon on my desktop was greeting me with a big red exclamation mark. Panic! What happened? Was my system infected by a virus? Exclamation marks normally mean: 'Danger!' so I searched and found that several users were affected. It pointed out that this exclamation marks just showed that the files marked were considered as 'just an icon (emblem) in your icon set and indicates that you haven't trusted those files' (tsjuan). *facepalm*

Another annoyance hit when suddenly the icons of some gtk applications appeared blurred in the panel e.g. the bluetooth manager icon and claws-mail icon (issue #945). After a while a patch was released which fixed the issue. But then after another update the icons were suddenly blurred again. I filed bug report issue #1864. That was when I first got in trouble with tsjuan who considered my lack of understanding for removing a fully working patch as a 'rant'.

Yesterday I suddenly found all my shortcuts defined in openbox configuration file lxqt-rc.xml out of function. I STFW for hours but found nothing about it, remembered a similar issue long ago when archlinux did not update all LXQt components at same time leaving the users with a partial update. But there was no hint to find anywhere using any search engine. The most irritating: As the shortcuts still worked for the one user on the system they did not for another. So I filed another bug report. 

Again it was tsjuan who closed it again in only three minutes after I've filed it telling me 'It's explained in several places.' Really?  'Sorry, I have no time to find them'. Well, me too! - Also I tried but had no luck 'but in short: We don't make an exception for Openbox anymore.' *grrr*

WTF? Separating the openbox config files into several instances for each DE which are using it was totally reasonable! Now you will not be able to use pure openbox and LXQt on one system anymore as they share the same rc.xml. 'We don't make an exception for Openbox anymore' is the same arrogant explanation as 'blame the gtk developers' for their non working icons. At least he was so gracious to point me to the LXQt release notes for 0.17.0.

But the thing there that upset me most was the crazy babble about 'Openbox configuration is removed for the sake of WM agnosticism'. WTF again?
"Agnosticism is the view that the existence of God, of the divine or the supernatural is not certainly known." What has that to do with 'for the sake of WM'? Yes, I had read the release notes and was wondering why obconf-qt (the openbox configuration) was still there! 
How should someone read from this out of touch with the real world nonsense that lxqt-rc.xml was deprecated and rc.xml is used again?

Perhaps the LXQt developers should focus their work on making the DE more useful and stable instead of keep removing useful features an add superfluous nonsense no one really needs. And please remember: As long as you claim to make LXQt for users and not for your personal satisfaction only, as long as there are user who really try hard to use LXQt in a productive way you should be more careful and communicate changes in a replicable way instead of some crypted philosophical statements. Not everyone is a native english speaker or a would-be philosopher.

And if you really want your users to file bug reports from which you could benefit to improve your project you should be a little bit friendlier at least - even when these reports base on misunderstanding or a lack of knowledge.

Thank you very much and have a nice day.

**Comment by stefonarch** [comment]
Nobody will ever do things right for everybody and features added or removed will always upset someone, but thanks for your feedback.

We expected some negative reactions after the  lxqt-rc.removal, but nothing is really lost - pure openbox session users still can have their different openbox settings in /usr/share/xsessions/openbox.desktop:
```
...
Exec=/usr/bin/openbox-session  --config-file /path/to/your/rc.xml
...
```
And "agnostic"[ isn't](https://www.merriam-webster.com/dictionary/agnostic) only a religous term. 
There are some conflicts with other WM too and the decision was not to prefer one.
